src/main/java/com/team701/buddymatcher/controllers/communication/CommunicationController.java

```diff
@@ -89,6 +89,18 @@ public CommunicationController(CommunicationService communicationService, UserSe
         }
     }
 
+
+    @Operation(summary = "Delete method to delete a user's conversation")
+    @DeleteMapping(path = "/messages/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> deleteUserMessages(@Parameter(hidden = true) @SessionAttribute("UserId") Long userId, @PathVariable("id") Long buddyId) {
+        try {
+            communicationService.deleteMessagesBetweenUsers(userId, buddyId);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (NoSuchElementException e) {
+          throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
```

Comment:
```suggestion
        } catch (NoSuchElementException e) {
          throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
```
NoSuchElementException will not be caught when the user is not found